sigle/src/modules/analytics/PublishedStoryItem.tsx

```diff
@@ -1,25 +1,52 @@
-import { ArrowRightIcon } from '@radix-ui/react-icons';
+import { ArrowLeftIcon, ArrowRightIcon } from '@radix-ui/react-icons';
 import { stagger } from 'motion';
 import { useMotionAnimate } from 'motion-hooks';
 import { useEffect } from 'react';
+import { useQuery } from 'react-query';
 import { styled } from '../../stitches.config';
 import { SubsetStory } from '../../types';
 import { Box, Flex, Text } from '../../ui';
+import { AnalyticsHistoricalResponse } from './stats/types';
+import { baseUrl, FATHOM_MAX_FROM_DATE } from './stats/utils';
 
 const StoryImage = styled('img', {
   objectFit: 'cover',
   objectPosition: 'center',
   width: 60,
   height: 43,
   br: '$1',
-  ml: '$2',
 });
 
+const testId = 'JA9dBfdPDp7kQhkFkgPdv';
+
+const fetchStoryViews = async (storyId: string) => {
+  const url = `${baseUrl}/api/analytics/historical?dateFrom=${FATHOM_MAX_FROM_DATE}&dateGrouping=month&storyId=${storyId}`;
+
+  const statsRes = await fetch(url);
+  const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+  let views;
+  if (statsData.stories.length > 0) {
+    views = statsData.stories[0].pageviews;
+  } else {
+    views = 0;
+  }
+  return views;
+};
+
 interface PublishedStoryItemProps {
   story: SubsetStory;
+  onClick: () => void;
+  individualStory?: boolean;
 }
 
-export const PublishedStoryItem = ({ story }: PublishedStoryItemProps) => {
+export const PublishedStoryItem = ({
+  story,
+  onClick,
+  individualStory = false,
+}: PublishedStoryItemProps) => {
+  const { data } = useQuery<number, Error>(['fetchStoryViews'], () =>
```

Comment:
here you are calling the same endpoint as the graph right? To avoid asking the server to do the computation twice, would using the same cache key with useQuery make only one call to the backend?

Reply:
checked the docs and yeah that should work thx :)